packages/daisy/src/components/pagination/pagination.tsx

```diff
@@ -0,0 +1,77 @@
+import { $, component$, PropFunction } from '@builder.io/qwik';
+import { Button } from '@qwik-ui/material';
+import { getPaginationItems } from '@qwik-ui/shared';
+
+export interface PaginationProps {
+  pages: number;
+  page: number;
+  onPaging$: PropFunction<(index: number) => void>;
+}
+
+/**
+ * Pagination
+ * ----------
+ * A pagination component
+ * first page is 1
+ *
+ * @example
+ * <Pagination pages={15} page={store.page} onPaging$={incrementCount} />
+ */
+export const Pagination = component$((props: PaginationProps) => {
+  const pagi = getPaginationItems(props.pages, props.page);
+
+  const _onPaging$ = $((page: number) => {
+    if (page < 1 || page > props.pages) return;
+    props.onPaging$(page);
+  });
+
+  return (
```

Comment:
With this approach we duplicated a lot of code, let's use the headless pagination component instead. The daisy one should be a simple wrapper headless wrapper with daisy styles

Reply:
Oh I see, we could use <Slot> to solve this case. The headless will receive two headless buttons, the daisy inject with slots the daisy ones.

Reply:
I see. The only solution here is modify state.page directly from your button and calculate all the props (disable, class) inside the buttons. For my experience, there is no way to use events on Slots